### PR TITLE
Adding support for mocha 6 stats collector.

### PR DESCRIPTION
**Pull request by passuied**
Major version. Requires peer dependency of mocha 6+

### CODE REVIEW

File: test/helpers/mock-runner.js [hunk 1]
```diff
@@ -2,10 +2,12 @@
 
 var EventEmitter = require('events').EventEmitter;
 var util = require('util');
+var createStatsCollector = require('mocha/lib/stats-collector');
 
 // mock test runner
 function Runner() {
   Runner.super_.call(this);
+  createStatsCollector(this);
```
Comment (john-ro) [inline test/helpers/mock-runner.js]: Can probably remove this now

File: index.js [hunk 2]
```diff
@@ -74,17 +75,20 @@ function getTags(testTitle) {
  * @param {Object} options - mocha options
  */
 function MochaXUnitReporter(runner, options) {
+  createStatsCollector(runner);
   this._options = configureDefaults(options);
   this._runner = runner;
+  // get functionality from the Base reporter
+  Base.call(this, runner);
+  var stats = this.stats;
```
Comment (john-ro) [inline index.js]: Not used anywhere

**Review by john-ro — APPROVED** [review]
Minor comments but this should be ready

File: test/helpers/mock-runner.js (hunk 1 above)
Comment (passuied) [reply test/helpers/mock-runner.js]: addressed

**Pull request merged**